vendor/bat-native-ads/src/bat/ads/internal/federated/log_entries/number_of_tabs_opened_in_past_30_minutes.cc

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ads/internal/federated/log_entries/number_of_tabs_opened_in_past_30_minutes.h"
+
+#include <cstdint>
+
+#include "bat/ads/ads_client.h"
+#include "bat/ads/internal/ads_client_helper.h"
+#include "bat/ads/internal/federated/covariate_logs_util.h"
+#include "bat/ads/internal/user_activity/user_activity.h"
+#include "bat/ads/pref_names.h"
+
+namespace ads {
+
+namespace {
+
+int64_t ComputeTabsOpened(const UserActivityEventList& events) {
+  const int64_t count = std::count_if(
+      events.cbegin(), events.cend(), [](const UserActivityEventInfo& event) {
+        return event.type == UserActivityEventType::kOpenedNewTab;
+      });
+
+  return count;
+}
```

Comment:
`tabs_opened_util.h` for this free function with tests

Reply:
Based on our DM, I think `user_activity_tabs_util.h/.cc`

Reply:
Fixed